viz/Apex-Radar/src/common.ts

```diff
@@ -0,0 +1,312 @@
+import {
+  ObjectFormat,
+  RowEntry,
+  ObjectRow,
+  FieldsByConfigId,
+  StyleById,
+} from '@google/dscc';
+import ApexCharts from 'apexcharts';
+
+interface FontInfo {
+  family: string;
+  size: number;
+  color: string;
+}
+interface Chart {
+  chart: ApexCharts;
+  element: Element;
+}
+let myChart: Chart = {
+  chart: undefined,
+  element: undefined,
+};
+const CHARTNAME = 'myChart';
+
+export function drawViz(data: ObjectFormat): void {
+  const newChart: boolean = document.getElementById(CHARTNAME) == undefined;
+  const tables = populateTables(data.tables.DEFAULT, data.fields);
+  const styling = populateStyle(data.style, tables.labels.length);
+
+  const options = {
+    series: tables.series,
+    labels: tables.labels,
+    tooltip: styling.tooltip,
+    chart: styling.chart,
+    fill: styling.fill,
+    markers: styling.markers,
+    dataLabels: styling.datalabels,
+    plotOptions: styling.plotoptions,
+    xaxis: styling.xaxis,
+    yaxis: styling.yaxis,
+    stroke: styling.stroke,
+    legend: styling.legend,
+  };
+  if (newChart) {
+    myChart.element = document.createElement('div');
+    myChart.element.id = CHARTNAME;
+    document.body.appendChild(myChart.element);
+    myChart.chart = new ApexCharts(myChart.element, options);
+    myChart.chart.render();
+    console.log('New chart created.');
+  } else {
+    myChart.chart.updateOptions(options);
+    console.log('Chart updated.');
+  }
+}
+/**
+ * Logic to translate Dims/Metrics into ApexSeries and ApexLabels
+ * @param vizData
+ * @param vizFields
+ */
+export function populateTables(
+  vizData: ObjectRow[],
+  vizFields: FieldsByConfigId
+) {
+  const metrics: (number)[][] = [];
+  const series: ApexAxisChartSeries = [];
+  const labels: (RowEntry)[] = [];
+  for (let i = 0; i < vizData.length; ++i) {
+    labels[i] = vizData[i].dimID[0];
+    for (let j = 0; j < vizData[0].metricID.length; j++) {
+      if (!metrics[j]) metrics[j] = [];
+      metrics[j][i] = Number(vizData[i].metricID[j]);
+    }
+  }
+  //create a 'Series' for each metric
+  for (let i = 0; i < metrics.length; ++i) {
+    const newSeries = {
+      name: vizFields.metricID[i].name,
+      data: metrics[i],
+    };
+    series.push(newSeries);
+  }
```

Comment:
Optional: JavaScript has some great functional programming methods for 1:1 translations like this - they read a bit better too (the comment wouldn't be necessary)

```suggestion
  const series = metrics.map((metric, i) => {
    return {
      name: vizFields.metricID[i].name,
      data: metric,
    };
  });
```

Up to you - personally, I think this is fine the way it is since the data is being generated from two parallel arrays (`vizFields.metricID` and `metrics`), but generally using `.map` / `.filter` / `.reduce` is less error-prone.